RedditSharp/Reddit.cs

```diff
@@ -546,12 +547,84 @@ public Comment GetComment(Uri uri)
         /// <param name="query">string to query</param>
         /// <param name="sortE">Order by <see cref="Sorting"/></param>
         /// <param name="timeE">Order by <see cref="TimeSorting"/></param>
+        /// <param name="author">The user who submitted the post</param>
+        /// <param name="flair">The text of the link flair on the post.</param>
+        /// <param name="nsfw">include NSFW posts.</param>
+        /// <param name="self">includ eText post.</param>
+        /// <param name="selfText">For self-posts, the body of the post.</param>
+        /// <param name="site">The domain of the submitted URL.</param>
+        /// <param name="subreddit">The submission's subreddit.</param>
+        /// <param name="title">The submission title.</param>
+        /// <param name="url">The submission's URL (the website's address)</param>
+        /// <remarks>https://www.reddit.com/wiki/search#wiki_field_search</remarks>
         /// <returns></returns>
-        public Listing<T> Search<T>(string query, Sorting sortE = Sorting.Relevance, TimeSorting timeE = TimeSorting.All) where T : Thing
+        public Listing<T> Search<T>(string query, 
+            Sorting sortE = Sorting.Relevance, 
+            TimeSorting timeE = TimeSorting.All, 
+            string author = null,
+            string flair = null,
+            bool? nsfw = null,
```

Comment:
What's the purpose/thought process behind this being nullable?

Reply:
I made it nullable because i need something for the default parameter.  I wanted something that for sure meant that it didn't need to be included in the query.